File: myfortune/Services/PremiumManager.swift
import Foundation

/// Tracks whether the user has purchased the "Remove Ads" upgrade.
///
/// This is a lightweight stub backed by `UserDefaults` so the ad-display
/// logic has something concrete to branch on today. Wiring this up to a
/// real StoreKit non-consumable purchase ("Remove Ads") is a separate task —
/// when that lands, `purchasePremium()`/`restorePurchases()` should be
/// replaced with real StoreKit transaction handling that calls
/// `setPremium(true)` on success.
final class PremiumManager {
    static let shared = PremiumManager()

    private let defaultsKey = "isPremiumUser"
    private let defaults: UserDefaults

    init(defaults: UserDefaults = .standard) {
        self.defaults = defaults
    }

    /// `true` once the user has purchased "Remove Ads". Defaults to `false`,
    /// meaning ads are shown until a purchase is recorded.
    var isPremium: Bool {
        defaults.bool(forKey: defaultsKey)
    }

    func setPremium(_ value: Bool) {
        defaults.set(value, forKey: defaultsKey)
    }
}

File: android/app/src/main/java/com/yourwish/cometrueaifortune/service/PremiumManager.kt
package com.yourwish.cometrueaifortune.service

import android.content.Context
import android.content.SharedPreferences

/**
 * Tracks whether the user has purchased the "Remove Ads" upgrade.
 *
 * Mirrors myfortune/Services/PremiumManager.swift: a lightweight stub backed by
 * SharedPreferences so the ad-display logic has something concrete to branch on
 * today. Wiring this up to a real Google Play Billing non-consumable purchase
 * ("Remove Ads") is a separate task — when that lands, `setPremium(true)` should
 * be called from the Billing purchase-acknowledgement callback.
 */
class PremiumManager(context: Context) {

    private val prefs: SharedPreferences =
        context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

    /** `true` once the user has purchased "Remove Ads". Defaults to `false`. */
    val isPremium: Boolean
        get() = prefs.getBoolean(KEY_IS_PREMIUM, false)

    fun setPremium(value: Boolean) {
        prefs.edit().putBoolean(KEY_IS_PREMIUM, value).apply()
    }

    companion object {
        private const val PREFS_NAME = "myfortune_premium"
        private const val KEY_IS_PREMIUM = "isPremiumUser"
    }
}
